zookeeper-server/src/main/java/org/apache/zookeeper/ClientCnxn.java

```diff
@@ -469,6 +470,16 @@ private static String makeThreadName(String suffix) {
         return name + suffix;
     }
 
+    /**
+     * Tests that current thread is the main event loop.
+     * This method is useful only for tests inside ZooKeeper project
+     * it is not a public API intended for use by external applications.
+     * @return true if Thread.currentThread() is an EventThread.
+     */
+    public static boolean isInEventThread() {
```

Comment:
just an idea: what about making it package private, and moving the testcase (or the whole test file) to the org.apache.zookeeper package?

Reply:
@symat 
I made it public because I thought that this kind of assertions would be very useful while working on the EventThread.
I also didn't want to move the test case, in order to limit the impact of the patch over the codebase.
I will do it if you feel strong

Reply:
Usually I try to avoid adding anything to the code "just for the tests". On the other hand this method can be even handy when the users are writing their own callbacks. Or for other unit tests. And the EventThread is an inner-class, so I wouldn't expose it either... 

So all-in-all I think this is a meaningful change, let's keep it.